ts/src/coincatch.ts

```diff
@@ -0,0 +1,5299 @@
+         * @param {float} [params.cost] *market buy only* the quote quantity that can be used as an alternative for the amount
+         * @param {bool} [params.postOnly] if true, the order will only be posted to the order book and not executed immediately
+         * @param {string} [params.timeInForce] 'GTC', 'IOC', 'FOK' or 'PO' (default 'GTC')
+         * @param {string} [params.clientOrderId] a unique id for the order (max length 40)
+         * @returns {object} request to be sent to the exchange
+         */
+        let methodName = 'createSpotOrderRequest';
+        // spot market info has no presicion so we do not use it
+        [ methodName, params ] = this.handleParamString (params, 'methodName', methodName);
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+            'side': side,
+            'orderType': type,
+        };
+        const isMarketOrder = (type === 'market');
+        const timeInForceAndParams = this.handleTimeInForceAndPostOnly (methodName, params, isMarketOrder);
+        params = timeInForceAndParams['params'];
+        const timeInForce = timeInForceAndParams['timeInForce'];
+        let cost: Str = undefined;
+        [ cost, params ] = this.handleParamString (params, 'cost');
+        let triggerPrice: Str = undefined;
+        [ triggerPrice, params ] = this.handleParamString (params, 'triggerPrice');
+        const isMarketBuy = isMarketOrder && (side === 'buy');
+        if ((!isMarketBuy) && (cost !== undefined)) {
+            throw new NotSupported (this.id + ' ' + methodName + ' supports cost parameter for market buy orders only');
+        }
+        if (isMarketBuy) {
+            const costAndParams = this.handleRequiresPriceAndCost (methodName, params, price, amount, cost);
+            cost = costAndParams['cost'];
+            params = costAndParams['params'];
+        }
+        if (triggerPrice === undefined) {
+            if (type === 'limit') {
+                request['price'] = price; // spot markets have no precision
+            }
+            request['quantity'] = isMarketBuy ? cost : amount.toString (); // spot markets have no precision
+            request['force'] = timeInForce ? timeInForce : 'normal'; // the exchange requres force but accepts any value
+        } else {
+            request['triggerPrice'] = triggerPrice; // spot markets have no precision
+            if (timeInForce !== undefined) {
+                request['timeInForceValue'] = timeInForce;
+            }
+            let clientOrderId: Str = undefined;
+            [ clientOrderId, params ] = this.handleParamString (params, 'clientOrderId');
+            if (clientOrderId !== undefined) {
+                request['clientOid'] = clientOrderId;
+            }
+            if (type === 'limit') {
+                request['executePrice'] = price; // spot markets have no precision
```

Comment:
don't we need to convert price to string as we do with amount? (if it works as is then it is ok)

Reply:
Yes, it works good